Source/WebKit/UIProcess/WebPageProxyInternals.h

```diff
@@ -0,0 +1,370 @@
+#endif
+
+#if HAVE(TOUCH_BAR)
+    TouchBarMenuData touchBarMenuData;
+#endif
+
+#if ENABLE(TOUCH_EVENTS)
+    TouchEventTracking touchEventTracking;
+#endif
+
+#if ENABLE(TOUCH_EVENTS) && !ENABLE(IOS_TOUCH_EVENTS)
+    Deque<QueuedTouchEvents> touchEventQueue;
+#endif
+
+#if ENABLE(TRACKING_PREVENTION)
+    MonotonicTime didFinishDocumentLoadForMainFrameTimestamp;
+#endif
+
+#if ENABLE(UI_SIDE_COMPOSITING)
+    VisibleContentRectUpdateInfo lastVisibleContentRectUpdate;
+#endif
+
+#if ENABLE(VIDEO_PRESENTATION_MODE)
+    RunLoop::Timer fullscreenVideoTextRecognitionTimer;
+    std::optional<PlaybackSessionContextIdentifier> currentFullscreenVideoSessionIdentifier;
+#endif
+
+#if ENABLE(WEBXR) && !USE(OPENXR)
+    std::unique_ptr<PlatformXRSystem> xrSystem;
+#endif
+
+    explicit Internals(WebPageProxy&);
+
+    SpeechSynthesisData& speechSynthesisData();
+
+    // WebPopupMenuProxy::Client
+    void valueChangedForPopupMenu(WebPopupMenuProxy*, int32_t newSelectedIndex) final;
+    void setTextFromItemForPopupMenu(WebPopupMenuProxy*, int32_t index) final;
+    NativeWebMouseEvent* currentlyProcessedMouseDownEvent() final;
+#if PLATFORM(GTK)
+    void failedToShowPopupMenu() final;
+#endif
+
+#if ENABLE(APPLE_PAY)
+    // WebPaymentCoordinatorProxy::Client
+    IPC::Connection* paymentCoordinatorConnection(const WebPaymentCoordinatorProxy&) final;
+    const String& paymentCoordinatorBoundInterfaceIdentifier(const WebPaymentCoordinatorProxy&) final;
+    const String& paymentCoordinatorSourceApplicationBundleIdentifier(const WebPaymentCoordinatorProxy&) final;
+    const String& paymentCoordinatorSourceApplicationSecondaryIdentifier(const WebPaymentCoordinatorProxy&) final;
+    void paymentCoordinatorAddMessageReceiver(WebPaymentCoordinatorProxy&, IPC::ReceiverName, IPC::MessageReceiver&) final;
+    void paymentCoordinatorRemoveMessageReceiver(WebPaymentCoordinatorProxy&, IPC::ReceiverName) final;
+#endif
+#if ENABLE(APPLE_PAY) && PLATFORM(IOS_FAMILY)
+    UIViewController *paymentCoordinatorPresentingViewController(const WebPaymentCoordinatorProxy&) final;
+    const String& paymentCoordinatorCTDataConnectionServiceType(const WebPaymentCoordinatorProxy&) final;
+    std::unique_ptr<PaymentAuthorizationPresenter> paymentCoordinatorAuthorizationPresenter(WebPaymentCoordinatorProxy&, PKPaymentRequest *) final;
+#endif
+#if ENABLE(APPLE_PAY) && PLATFORM(IOS_FAMILY) && ENABLE(APPLE_PAY_REMOTE_UI_USES_SCENE)
+    void getWindowSceneIdentifierForPaymentPresentation(WebPageProxyIdentifier, CompletionHandler<void(const String&)>&&) final;
+#endif
+#if ENABLE(APPLE_PAY) && PLATFORM(MAC)
+    NSWindow *paymentCoordinatorPresentingWindow(const WebPaymentCoordinatorProxy&) final;
+#endif
+
+#if ENABLE(INPUT_TYPE_COLOR)
+    // WebColorPickerClient
+    void didChooseColor(const WebCore::Color&) final;
+    void didEndColorPicker() final;
+#endif
+
+#if PLATFORM(MACCATALYST)
+    // EndowmentStateTracker::Client
+    void isUserFacingChanged(bool) final;
```

Comment:
Looks like this may have broken the build:
```
Source/WebKit/UIProcess/WebPageProxy.h:1384:36: error: only virtual member functions can be marked 'final'
    void isUserFacingChanged(bool) final;
                                   ^~~~~
1 error generated.
```